### PR TITLE
PLT-7787 Make 'version' optional in Testing form

### PR DESCRIPTION
**Pull request by amnambiar**
https://input-output.atlassian.net/browse/PLT-7787

* Make 'Version', 'Subject' optional in Testing form (in dapp within /profile/current) - either not present or null (not empty string)
* If Subject is not available in profile dapp, then show field 'Subject' in 'Review Certification Metadata' form (as it is mandatory to generate off-chain on-chain metadatas)

BE - `chore/make-version-optional` [PR #109](https://github.com/input-output-hk/dapps-certification/pull/109) -> currently in master

## Summary by CodeRabbit

New Feature:
- Introduced a more flexible form handling in the certification process. The `version` and `subject` fields are now optional, providing users with more flexibility when filling out forms.

Improvement:
- Replaced the `Toast` component with a `Snackbar` component for error messages, enhancing the user experience with more modern and intuitive notifications.

New Feature:
- Added a new action to generate a run certificate, improving the certification process.

Improvement:
- Updated the profile handling logic to better manage user data and improve the overall user experience.

Refactor:
- Streamlined the `CertificationMetadataForm` component, improving code readability and maintainability.

### CODE REVIEW

**Review by EehMauro — APPROVED** [review]

File: src/components/CertificationMetadataForm/index.tsx [hunk 1]
```diff
@@ -58,6 +58,7 @@ const CertificationMetadataForm = (props: Props) => {
         ...request,
         certificationLevel: !props.isReviewCertification ? parseInt(request.certificationLevel) : undefined,
         report: !props.isReviewCertification ? request.report.map(report => report.value) : undefined,
+        subject: props.isReviewCertification && profile?.dapp?.subject ? undefined : request.subject
```
Comment (bogdan-manole) [inline src/components/CertificationMetadataForm/index.tsx]: Can you elaborate a bit on the condition used here (preferably with a comment in the code)? 
Why If there is a subject in the `Dapp` profile we use `undefined` for the request and only when there is nothing we use what the user filled in?

Comment (bogdan-manole) [inline src/components/CertificationMetadataForm/index.tsx]: In some places, we have `null` for empty object properties and here we have `undefined`. Is there a specific reason for that?

**Review by bogdan-manole — CHANGES_REQUESTED** [review]

File: src/components/CertificationMetadataForm/index.tsx (hunk 1 above)
Comment (amnambiar) [reply src/components/CertificationMetadataForm/index.tsx:61]: Here we tried to use the same CertificationMetadataForm component for 'Auditor Report Upload' and 'Review Certification Metadata'

Comment (amnambiar) [reply src/components/CertificationMetadataForm/index.tsx:61]: This is just to indicate that the properties with undefined, need not be present in the UI to accept values from the user. So, if _isReviewCertification_, then certificationLevel, report and subject will not be present.

Comment (amnambiar) [reply src/components/CertificationMetadataForm/index.tsx]: This was done with the intent to not include the property in the request. I've updated it to a different piece of code now

Comment (amnambiar) [reply src/components/CertificationMetadataForm/index.tsx]: This was done with the intent to not include the property in the request. I've updated it to a different piece of code now

**Comment by amnambiar** [comment]
@bogdan-manole , passing over to you to have a re-look at this PR. Thanks,

**Comment by amnambiar** [comment]
@bogdan-manole , requesting re-review of this PR

**Review by bogdan-manole — APPROVED** [review]
tested and it's working

**Pull request merged**